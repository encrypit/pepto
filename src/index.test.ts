import { digest } from './index';

describe('digest', () => {
  it.each(['SHA-1', 'SHA-256', 'SHA-384', 'SHA-512'] as const)(
    'hashes message with algorithm "%s"',
    async (algorithm) => {
      expect(await digest(algorithm, '')).toMatchSnapshot();
    },
  );
});
