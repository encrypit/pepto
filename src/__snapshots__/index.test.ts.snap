// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`digest hashes message with algorithm "SHA-1" 1`] = `"da39a3ee5e6b4b0d3255bfef95601890afd80709"`;

exports[`digest hashes message with algorithm "SHA-256" 1`] = `"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"`;

exports[`digest hashes message with algorithm "SHA-384" 1`] = `"38b060a751ac96384cd9327eb1b1e36a21fdb71114be07434c0cc7bf63f6e1da274edebfe76f65fbd51ad2f14898b95b"`;

exports[`digest hashes message with algorithm "SHA-512" 1`] = `"cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e"`;
